unittests/compiled_tests.cpp

```diff
@@ -352,6 +352,11 @@ TEST_CASE("Functor cast")
   CHECK(d == 3 * 6);
 }
 
+TEST_CASE("BOM at beginning of string")
+{
+  chaiscript::ChaiScript_Basic chai(create_chaiscript_stdlib(),create_chaiscript_parser());
+  CHECK_THROWS_AS(chai.eval<std::string>("\xef\xbb\xbfprint \"Hello World\""), chaiscript::exception::eval_error);
+}
```

Comment:
Still needs a test case for other binary/non-ANSI garbage in the input at random positions (beginning, middle, end).

Reply:
I left just one non-ASCII character and moved it to the middle of the string.